src/overlays/actors/ovl_En_Ms/z_en_ms.c

```diff
@@ -28,38 +32,159 @@ const ActorInit En_Ms_InitVars = {
+    {
+        COLTYPE_NONE,
+        AT_NONE,
+        AC_ON | AC_TYPE_PLAYER,
+        OC1_ON | OC1_TYPE_ALL,
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK0,
+        { 0x00000000, 0x00, 0x00 },
+        { 0xF7CFFFFF, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_ON,
+        OCELEM_ON,
+    },
     { 22, 37, 0, { 0, 0, 0 } },
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80952BCC[] = {
+static InitChainEntry sInitChain[] = {
     ICHAIN_U8(targetMode, 2, ICHAIN_CONTINUE),
     ICHAIN_F32(targetArrowOffset, 500, ICHAIN_STOP),
 };
 
-#endif
-
-extern ColliderCylinderInit D_80952BA0;
-extern InitChainEntry D_80952BCC[];
-
-extern UNK_TYPE D_060005EC;
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/EnMs_Init.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/EnMs_Destroy.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/func_80952734.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/func_809527F8.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/func_809529AC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/func_80952A1C.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/EnMs_Update.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Ms/EnMs_Draw.s")
+extern AnimationHeader D_060005EC;
+extern FlexSkeletonHeader D_06003DC0;
+
+void EnMs_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnMs* this = THIS;
+
+    Actor_ProcessInitChain(thisx, sInitChain);
+    SkelAnime_InitFlex(globalCtx, &this->skelAnime, &D_06003DC0, &D_060005EC, this->jointTable, this->morphTable, 9);
+    Collider_InitCylinder(globalCtx, &this->collider);
+    Collider_SetCylinderType1(globalCtx, &this->collider, &this->actor, &sCylinderInit);
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 35.0f);
+    Actor_SetScale(&this->actor, 0.015f);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE; // Eating Magic Beans all day will do that to you
+    this->actionFunc = EnMs_Wait;
+    this->actor.speedXZ = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->actor.gravity = -1.0f;
+}
+
+void EnMs_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnMs* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->collider);
+}
+
+void EnMs_Wait(EnMs* this, GlobalContext* globalCtx) {
+    s16 yawDiff = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+
+    if (gSaveContext.inventory.items[SLOT_MAGIC_BEANS] == ITEM_NONE) {
+        this->actor.textId = 0x92E; // "[...] You're the first customer [...]"
+    } else {
+        this->actor.textId = 0x932; // "[...] So you liked my Magic Beans [...]"
+    }
+
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
+        this->actionFunc = EnMs_Talk;
+        return;
+    }
+
+    if ((this->actor.xzDistToPlayer < 90.0f) && (ABS_ALT(yawDiff) < 0x2000)) {
+        func_800B8614(&this->actor, globalCtx, 90.0f);
+    }
```

Comment:
```suggestion
    if (func_800B84D0(&this->actor, globalCtx) != 0) {
        this->actionFunc = EnMs_Talk;
    } else if ((this->actor.xzDistToPlayer < 90.0f) && (ABS_ALT(yawDiff) < 0x2000)) {
        func_800B8614(&this->actor, globalCtx, 90.0f);
    }
```
Possible `else if` to remove the early return?

Reply:
I remain in two minds about this. I think it's nice to do it where possible: saves indentation, makes it clear that setting the actionFunc is the last thing that happens. But we get cases where it must and can't be done.

Reply:
Personally I prefer the `else if`, but I'll leave it up to you.